activesupport/lib/active_support/current_attributes.rb

```diff
@@ -164,16 +164,18 @@ def current_instances_key
         end
 
         def method_missing(name, ...)
-          # Caches the method definition as a singleton method of the receiver.
-          #
-          # By letting #delegate handle it, we avoid an enclosure that'll capture args.
-          singleton_class.delegate name, to: :instance
-
-          send(name, ...)
+          instance.public_send(name, ...)
         end
 
         def respond_to_missing?(name, _)
-          super || instance.respond_to?(name)
+          instance.respond_to?(name) || super
+        end
```

Comment:
@byroot I was able to make this change and pass the suite:

```diff
diff --git a/activesupport/lib/active_support/current_attributes.rb b/activesupport/lib/active_support/current_attributes.rb
index 4ffa1edd5c..5f751f2d27 100644
--- a/activesupport/lib/active_support/current_attributes.rb
+++ b/activesupport/lib/active_support/current_attributes.rb
@@ -172,14 +172,6 @@ def current_instances_key
           @current_instances_key ||= name.to_sym
         end
 
-        def method_missing(name, ...)
-          instance.public_send(name, ...)
-        end
-
-        def respond_to_missing?(name, _)
-          instance.respond_to?(name) || super
-        end
-
         def method_added(name)
           return if name == :initialize
           return unless public_method_defined?(name)
```

Is there a reason we retained those methods?

Reply:
> Is there a reason we retained those methods?

Mostly because I'm not 100% confident `method_added` cover 100% of the cases. I think it does, but I still have a small doubt.